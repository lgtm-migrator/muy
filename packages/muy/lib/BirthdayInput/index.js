import React from "react"
import TextField from "@material-ui/core/TextField"

export const getDateByAge = age => {
  const date = new Date(Date.now())
  const year = date.getFullYear() - age
  const month = `0${date.getUTCMonth()}`.slice(-2)
  const day = `0${date.getDate()}`.slice(-2)
  return `${year}-${month}-${day}`
}

const BirthdayInput = ({ minAge, maxAge, ...props }) => {
  return (
    <TextField
      inputProps={{
        max: maxAge && getDateByAge(maxAge),
        min: minAge && getDateByAge(minAge),
      }}
      type="date"
      {...props}
    />
  )
}

export default BirthdayInput
