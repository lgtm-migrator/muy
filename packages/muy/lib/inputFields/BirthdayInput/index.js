import React from "react"
import PropTypes from "prop-types"
import TextField from "@material-ui/core/TextField"

export const getDateByAge = (age) => {
  const date = new Date(Date.now())
  const year = date.getFullYear() - age
  const month = `0${date.getUTCMonth()}`.slice(-2)
  const day = `0${date.getDate()}`.slice(-2)
  return `${year}-${month}-${day}`
}

const BirthdayInput = ({ maxAge, minAge, ...props }) => {
  return (
    <TextField
      inputProps={{
        max: maxAge && getDateByAge(maxAge),
        min: minAge && getDateByAge(minAge),
      }}
      type="date"
      {...props}
    />
  )
}

export default BirthdayInput

BirthdayInput.propTypes = {
  maxAge: PropTypes.number,
  minAge: PropTypes.number,
}
