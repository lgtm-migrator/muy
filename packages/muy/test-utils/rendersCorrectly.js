import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
