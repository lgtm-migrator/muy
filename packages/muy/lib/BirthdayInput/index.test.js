import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import BirthdayInput from "./index"

describe("<BirthdayInput /> component", () => {
  describe("should render correctly", () => {
    beforeAll(() => {
      Date.now = jest.fn(() => 1482363367071)
    })
    it("with no props", () => {
      rendersCorrectly(<BirthdayInput />)
    })
    it("with minAge prop", () => {
      rendersCorrectly(<BirthdayInput minAge={18} />)
    })
    it("with maxAge prop", () => {
      rendersCorrectly(<BirthdayInput maxAge={100} />)
    })
    it("with both maxAge and minAge props", () => {
      rendersCorrectly(<BirthdayInput minAge={18} maxAge={100} />)
    })
  })
})
