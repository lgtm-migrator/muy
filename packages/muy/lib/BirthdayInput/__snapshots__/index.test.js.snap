// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BirthdayInput /> component should render correctly with both maxAge and minAge props 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      max="1916-11-21"
      min="1998-11-21"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="date"
    />
  </div>
</div>
`;

exports[`<BirthdayInput /> component should render correctly with maxAge prop 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      max="1916-11-21"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="date"
    />
  </div>
</div>
`;

exports[`<BirthdayInput /> component should render correctly with minAge prop 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      min="1998-11-21"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="date"
    />
  </div>
</div>
`;

exports[`<BirthdayInput /> component should render correctly with no props 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="date"
    />
  </div>
</div>
`;
